packages/cubejs-server-core/test/unit/RefreshScheduler.test.ts

```diff
@@ -234,6 +233,16 @@ cube('Bar', {
   ]),
 };
 
+function createDeferred() {
```

Comment:
We already have a similar functionality `createPromiseLock` in `@cubejs-backend/shared`;